### PR TITLE
Fix seg fault and enable parameter error=0 in FrechetShortcut

### PR DESCRIPTION
**Pull request by isivigno**
# PR Description

Fix seg fault due to recent compilers in FrechetShortcut (Bertrand Kerautret, Isabelle Sivignon)
Fix FrechetShortcut to enable the parameter error to be equal to 0 and add new tests in testFrechetShortcut (Isabelle Sivignon)

# Checklist

- [ ] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [ ] Doxygen documentation of the code completed (classes, methods, types, members...)
- [ ] Documentation module page added or updated.
- [ ] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [ ] No warning raised in Debug mode.
- [ ] All continuous integration tests pass (Github Actions)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -87,13 +87,16 @@
...
 
 - *Geometry package*
   - Fix Issue #1676 in testStabbingCircleComputer (Tristan Roussillon,
-    [#1688](https://github.com/DGtal-team/DGtal/pull/1688)
+	[#1688](https://github.com/DGtal-team/DGtal/pull/1688))
   - Fix const attribute that shouldn't be in FreemanChain (Colin Weill--Duflos,
-    [#1723](https://github.com/DGtal-team/DGtal/pull/1723))
+	[#1723](https://github.com/DGtal-team/DGtal/pull/1723))
+  - Fix seg fault due to recent compilers in FrechetShortcut (Bertrand Kerautret, Isabelle Sivignon)
+  - Fix FrechetShortcut to enable the parameter error to be equal to 0 and add new tests in testFrechetShortcut (Isabelle Sivignon)
```
Comment (dcoeurjo) [inline ChangeLog.md:98]: could you please add the link to the PR? thx

File: src/DGtal/geometry/curves/FrechetShortcut.h [hunk 2]
```diff
@@ -52,6 +52,9 @@
 
 #include "DGtal/geometry/curves/SegmentComputerUtils.h"
 
+
+#define PRECISION 0.00000001
```
Comment (dcoeurjo) [inline src/DGtal/geometry/curves/FrechetShortcut.h:56]: Would it be possible to have this as a constructor parameter (with the same default value) ?

**Review by dcoeurjo — COMMENTED** [review]
Thanks for the PR, I just have minor comments. best

**Comment by dcoeurjo** [comment]
Hi @isivigno, we are preparing the 1.4 release of DGtal. Could you please have a look to my comments on your PR?
thanks

File: ChangeLog.md (hunk 1 above)
Comment (kerautret) [reply ChangeLog.md]: I just do It in the merge

**Comment by dcoeurjo** [comment]
few minor errors in the CI bots. could you please have a look @kerautret ?
(I would like to include this PR in the release)

**Comment by kerautret** [comment]
> few minor errors in the CI bots. could you please have a look @kerautret ? (I would like to include this PR in the release)

Yes I am adding the PRECISION in the constructor (but also used in static)

**Comment by kerautret** [comment]
@dcoeurjo for the CI I added a PR on the @isivigno branches https://github.com/isivigno/DGtal/pull/2

**Comment by dcoeurjo** [comment]
I was able to merge he PR on @isivigno repo but the PR was against the master, not the freshet branch. Let me check

**Comment by kerautret** [comment]
for the precision constant I can finish the stashed starting inclusion on classes if you think it better

**Comment by dcoeurjo** [comment]
let see if this one passes all tests.

**Comment by kerautret** [comment]
> I was able to merge he PR on @isivigno repo but the PR was against the master, not the freshet branch. Let me check

oupst sure ;)

**Comment by dcoeurjo** [comment]
If you have anything you would like to change on this class, let's do that in the 1.5beta.

thanks

**Pull request merged**